creator-node/src/services/stateMachineManager/stateMonitoring/findReplicaSetUpdates.jobProcessor.js

```diff
@@ -0,0 +1,275 @@
+          }. Marking replica as unhealthy.`
+        )
+        unhealthyReplicas.add(secondary)
+
+        // Error case 2 - already marked unhealthy
+      } else if (unhealthyPeers.has(secondary)) {
+        logger.error(
+          `_findReplicaSetUpdatesForUser(): Secondary ${secondary} for user ${wallet} in unhealthy peer set. Marking replica as unhealthy.`
+        )
+        unhealthyReplicas.add(secondary)
+
+        // Error case 3 - low user sync success rate
+      } else if (
+        failureCount >= minFailedSyncRequestsBeforeReconfig &&
+        successRate < minSecondaryUserSyncSuccessPercent
+      ) {
+        logger.error(
+          `_findReplicaSetUpdatesForUser(): Secondary ${secondary} for user ${wallet} has userSyncSuccessRate of ${successRate}, which is below threshold of ${minSecondaryUserSyncSuccessPercent}. ${successCount} Successful syncs vs ${failureCount} Failed syncs. Marking replica as unhealthy.`
+        )
+        unhealthyReplicas.add(secondary)
+      }
+    }
+
+    /**
+     * If any unhealthy replicas found for user, enqueue an updateReplicaSetOp for later processing
+     */
+    if (unhealthyReplicas.size > 0) {
+      requiredUpdateReplicaSetOps.push({ ...user, unhealthyReplicas })
+    }
+
+    /**
+     * If this node is secondary for user, check both secondaries for health and enqueue SyncRequests against healthy secondaries
+     * Ignore unhealthy secondaries for now
+     */
+  } else {
+    // filter out false-y values to account for incomplete replica sets and filter out the
+    // the self node
+    replicaSetNodesToObserve = [
+      { endpoint: primary, spId: primarySpID },
+      ...replicaSetNodesToObserve
+    ]
+    replicaSetNodesToObserve = replicaSetNodesToObserve.filter((entry) => {
+      return entry.endpoint && entry.endpoint !== thisContentNodeEndpoint
+    })
+
+    for (const replica of replicaSetNodesToObserve) {
+      // If the map's spId does not match the query's spId, then regardless
+      // of the relationship of the node to the user, issue a reconfig for that node
+      if (
+        CNodeToSpIdMapManager.getCNodeEndpointToSpIdMap()[replica.endpoint] !==
+        replica.spId
+      ) {
+        unhealthyReplicas.add(replica.endpoint)
+      } else if (unhealthyPeers.has(replica.endpoint)) {
+        // Else, continue with conducting extra health check if the current observed node is a primary, and
+        // add to `unhealthyReplicas` if observed node is a secondary
+        let addToUnhealthyReplicas = true
+
+        if (replica.endpoint === primary) {
+          addToUnhealthyReplicas = !(await CNodeHealthManager.isPrimaryHealthy(
+            primary
+          ))
+        }
+
+        if (addToUnhealthyReplicas) {
+          unhealthyReplicas.add(replica.endpoint)
+        }
+      }
+    }
+
+    if (unhealthyReplicas.size > 0) {
+      requiredUpdateReplicaSetOps.push({ ...user, unhealthyReplicas })
+    }
+  }
+
+  return requiredUpdateReplicaSetOps
+}
```

Comment:
this is extracted from snapback's [_aggregateOps()](https://github.com/AudiusProject/audius-protocol/blob/master/creator-node/src/snapbackSM/snapbackSM.js#L1488-L1612)